src/antelope/stores/nfts.ts

```diff
@@ -41,6 +53,57 @@ export const useNftsStore = defineStore(store_name, {
         loggedInventory: state => state.__inventory['logged']?.list,
         loggedInventoryLoading: state => state.__inventory['logged']?.loading,
         getInventory: state => (label: string) => state.__inventory[label],
+        getUserFilter: state => state.__user_filter,
+        getCollectionList: state => (label: string): NftCollectionListItem[] => {
+            // used to populate the 'collections' dropdown in the NFT inventory page
+            const inventory = state.__inventory[label]?.list;
+            if (!inventory) {
+                return [];
+            }
+            const collection_list: NftCollectionListItem[] = [];
+            inventory.forEach((nft) => {
+                const collection = collection_list.find(c => c.contract === nft.contractAddress);
+                if (!collection) {
+                    collection_list.push({
+                        contract: nft.contractAddress,
+                        name: nft.contractPrettyName || truncateAddress(nft.contractAddress),
+                    });
+                }
+            });
+            // sort collection list such that items whose name begins with 0x are at the bottom, and those with names are at the top sorted alphabetically
+            collection_list.sort((a, b) => {
+                if (a.name.startsWith('0x') && !b.name.startsWith('0x')) {
+                    return 1;
+                } else if (!a.name.startsWith('0x') && b.name.startsWith('0x')) {
+                    return -1;
+                } else {
+                    return a.name.localeCompare(b.name);
+                }
+            });
+            return collection_list;
+        },
+        getUserFilteredInventory:  state => (label: string) => {
+            let inventory = state.__inventory[label]?.list ?? [];
+            const filter = state.__user_filter;
+
+            if (filter.collection) {
+                inventory = inventory.filter(nft => nft.contractAddress === filter.collection);
+            }
+
+            if (filter.searchTerm) {
+                const searchTermLower = filter.searchTerm.toLowerCase();
+                inventory = inventory.filter(nft => nft.name.toLowerCase().includes(searchTermLower) ||
+                        nft.id.toLowerCase().includes(searchTermLower) ||
+                        nft.description?.toLowerCase().includes(searchTermLower) ||
+                        nft.attributes.some(
+                            attr => [attr.label?.toLowerCase(), attr.text?.toLowerCase()].some(
+                                text => text?.includes(searchTermLower),
+                            ),
+                        ));
+            }
```

Comment:
Here you can add the check on the Collection Name too.

If I search for "Mino" it should return all "Mino NFT" collection items (currently there are 2 but none is returned)
Another example word can be "Vodka" for "MintedVodka OptiDeal" collection items

Reply:
yeah this was basically just the requirement. the collection dropdown filters for collection, the search bar is for everything else. i'll ask julio what he thinks